save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/testsuiteselector/TestSuiteSelector.kt

```diff
@@ -0,0 +1,183 @@
+                        asDynamic()["data-dismiss"] = "modal"
+                        ariaLabel = "Close"
+                        fontAwesomeIcon(icon = faTimesCircle)
+                        onClick = {
+                            onCancel()
+                        }
+                    }
+                }
+
+                div {
+                    className = ClassName("modal-body")
+                    testSuiteSelector {
+                        this.onTestSuiteIdUpdate = onTestSuiteIdUpdate
+                        this.preselectedTestSuiteIds = preselectedTestSuiteIds
+                    }
+                }
+
+                div {
+                    className = ClassName("modal-footer")
+                    div {
+                        className = ClassName("d-flex justify-content-center")
+                        button {
+                            type = ButtonType.button
+                            className = ClassName("btn btn-primary mt-4")
+                            +"Apply"
+                            onClick = {
+                                onSubmit()
+                            }
+                        }
+                    }
+                    div {
+                        className = ClassName("d-flex justify-content-center")
+                        button {
+                            type = ButtonType.button
+                            className = ClassName("btn btn-secondary mt-4")
+                            +"Cancel"
+                            onClick = {
+                                onCancel()
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+private fun ChildrenBuilder.buildButton(icon: FontAwesomeIconModule, tooltipText: String, onClickFun: () -> Unit) {
+    button {
+        type = ButtonType.button
+        title = tooltipText
+        className = ClassName("btn btn-outline-secondary")
+        fontAwesomeIcon(icon = icon)
+        onClick = {
+            onClickFun()
+        }
+
+        val jquery = kotlinext.js.require("jquery")
+        kotlinext.js.require("popper.js")
+        kotlinext.js.require("bootstrap")
+        asDynamic()["data-toggle"] = "tooltip"
+        asDynamic()["data-placement"] = "bottom"
+        jquery("[data-toggle=\"tooltip\"]").tooltip()
+    }
+}
+
+private fun testSuiteSelector() = FC<TestSuiteSelectorProps> { props ->
+    val (currentMode, setCurrentMode) = useState(if (props.preselectedTestSuiteIds.isEmpty()) {
+        TestSuiteSelectorMode.BROWSER
+    } else {
+        TestSuiteSelectorMode.MANAGER
+    })
+    div {
+        className = ClassName("d-flex align-self-center justify-content-around mb-2")
+        buildButton(faAlignJustify, "Manage linked test suites") { setCurrentMode(TestSuiteSelectorMode.MANAGER) }
+        buildButton(faPlus, "Browse public test suites") { setCurrentMode(TestSuiteSelectorMode.BROWSER) }
```

Comment:
It's not really clear what's the difference between these two, even from your video. This is a selector, why does it have managing mode?

Reply:
managing mode is needed to get information about selected test suites. Let's say you want to double-check what is selected to be 100% sure there is nothing odd. In this way, it might be good to have `managing mode`.

Reply:
Tried to show in video how it is planned to be used: I added c++ test suite by accident and it was very easy for me to notice that and remove it

Reply:
Ok, I see. Maybe we should highlight the button of the currently selected mode?

Reply:
Ok, will do it

Reply:
Done